rrtmgp/kernels/mo_reorder_kernels.F90

```diff
@@ -24,18 +24,34 @@ subroutine reorder_123x312_kernel(d1, d2, d3, array_in, array_out) &
     real(wp), dimension(d1, d2, d3), intent( in) :: array_in
     real(wp), dimension(d3, d1, d2), intent(out) :: array_out
 
-    integer :: i1, i2, i3
+    integer :: i1, i2, i3, i10, i30, i1diff, i3diff
+    integer, parameter :: tile = 32
```

Comment:
This is a little mysterious and magic-numbery. Can you add comments to explain the reasoning?

Reply:
This is basic blocking for more efficient cache usage, should also help on CPU. The threadblock needs to work on a small tile of the array such that non-contiguous reads (along the second or third array dimension) are effectively cached.
However, a better way to do the reordering is to use some CPU and GPU specific libraries.

Reply:
I realize this is what you're doing but since it's a pattern that might not be crystal-clear to naive readers, so I'm hoping you'll add comments to the code explaining.

Reply:
Oh, sorry - here I meant the specific choice of 32 as the tile size. The comments about the pattern apply to the code below.

Reply:
32 just empirically the best choice for the GPU. I can definitely add comments here